install/ui/src/freeipa/plugins/certmap.js

```diff
@@ -0,0 +1,394 @@
+// Copyright (C) 2017  FreeIPA Contributors see COPYING for license
+//
+
+
+define([
+        'dojo/_base/lang',
+        'dojo/_base/declare',
+        'dojo/Evented',
+        'dojo/on',
+        '../navigation',
+        '../field',
+        '../ipa',
+        '../phases',
+        '../reg',
+        '../widget',
+        '../text',
+        '../util',
+        // plain imports
+        '../search',
+        '../entity'],
+            function(lang, declare, Evented, on, navigation, mod_field, IPA,
+                     phases, reg, widget_mod, text, util) {
+/**
+ * Certificate map module
+ * @class
+ */
+var certmap = IPA.certmap = {
+
+    search_facet_group: {
+        facets: {
+            certmaprule_search: 'certmaprule_search',
+            domainlevel: 'domainlevel_details'
+        }
+    }
+};
+
+var make_certmaprule_spec = function() {
+return {
+    name: 'certmaprule',
+    facets: [
+        {
+            $type: 'search',
+            always_request_members: true,
+            details_facet: 'details',
+            facet_groups: [certmap.search_facet_group],
+            facet_group: 'search',
+            row_enabled_attribute: 'ipaenabledflag',
+            columns: [
+                'cn',
+                {
+                    name: 'ipaenabledflag',
+                    label: '@i18n:status.label',
+                    formatter: 'boolean_status'
+                },
+                'description'
+            ],
+            actions: [
+                'batch_disable',
+                'batch_enable'
+            ],
+            control_buttons: [
+                {
+                    name: 'disable',
+                    label: '@i18n:buttons.disable',
+                    icon: 'fa-minus'
+                },
+                {
+                    name: 'enable',
+                    label: '@i18n:buttons.enable',
+                    icon: 'fa-check'
+                }
+            ]
+        },
+        {
+            $type: 'details',
+            disable_facet_tabs: true,
+            facet_groups: [certmap.search_facet_group],
+            facet_group: 'search',
+            actions: [
+                'enable',
+                'disable',
+                'delete'
+            ],
+            header_actions: ['enable', 'disable', 'delete'],
+            state: {
+                evaluators: [
+                    {
+                        $factory: IPA.enable_state_evaluator,
+                        field: 'ipaenabledflag'
+                    }
+                ]
+            },
+            sections: [
+                {
+                    name: 'details',
+                    fields: [
+                        'cn',
+                        {
+                            $type: 'textarea',
+                            name: 'description'
+                        },
+                        'ipacertmapissuer',
+                        'ipacertmapmaprule',
+                        'ipacertmapmatchrule',
```

Comment:
Would be good if mapping and matching rules had examples written near the fields. Without reading any docs I don't have a clue what should be written there. Same in details page.